music_assistant/providers/spotify/__init__.py

```diff
@@ -1,1030 +1,888 @@
-            if response.status == 401:
-                self._auth_info = None
-                raise ResourceTemporarilyUnavailable("Token expired", backoff_time=0.05)
-            # handle temporary server error
-            if response.status in (502, 503):
-                raise ResourceTemporarilyUnavailable(backoff_time=30)
-            response.raise_for_status()
-            return await response.json(loads=json_loads)
-
-    def _fix_create_playlist_api_bug(self, playlist_obj: dict[str, Any]) -> None:
-        """Fix spotify API bug where incorrect owner id is returned from Create Playlist."""
-        if playlist_obj["owner"]["id"] != self._sp_user["id"]:
-            playlist_obj["owner"]["id"] = self._sp_user["id"]
-            playlist_obj["owner"]["display_name"] = self._sp_user["display_name"]
-        else:
-            self.logger.warning(
-                "FIXME: Spotify have fixed their Create Playlist API, this fix can be removed."
-            )
+"""Spotify musicprovider support for MusicAssistant with Podcast Support -Modular Implementation."""
+
+from __future__ import annotations
+
+import asyncio
+import os
+import time
+from collections.abc import AsyncGenerator
+from typing import TYPE_CHECKING, Any, cast
+
+from music_assistant_models.enums import (
+    ImageType,
+    MediaType,
+    ProviderFeature,
+)
+from music_assistant_models.errors import (
+    LoginFailed,
+    MediaNotFoundError,
+    ResourceTemporarilyUnavailable,
+    SetupFailedError,
+)
+from music_assistant_models.media_items import (
+    Album,
+    Artist,
+    ItemMapping,
+    MediaItemImage,
+    MediaItemType,
+    Playlist,
+    Podcast,
+    PodcastEpisode,
+    ProviderMapping,
+    SearchResults,
+    Track,
+    UniqueList,
+)
+
+from music_assistant.controllers.cache import use_cache
+from music_assistant.helpers.app_vars import app_var  # type: ignore[attr-defined]
+from music_assistant.helpers.json import json_loads
+from music_assistant.helpers.process import check_output
+from music_assistant.helpers.throttle_retry import ThrottlerManager, throttle_with_retries
+from music_assistant.helpers.util import lock
+from music_assistant.models.music_provider import MusicProvider
+
+from .constants import (
+    BASE_FEATURES,
+    CACHE_CATEGORY_EPISODES,
+    CACHE_CATEGORY_OTHER,
+    CACHE_CATEGORY_PODCASTS,
+    CACHE_CATEGORY_RECOMMENDATIONS,
+    CACHE_KEY_USER_INFO,
+    CONF_CLIENT_ID,
+    CONF_ENABLE_PODCASTS,
+    CONF_PLAYED_THRESHOLD,
+    CONF_REFRESH_TOKEN,
+    CONF_SYNC_PLAYED_STATUS,
+    LIKED_SONGS_FAKE_PLAYLIST_ID_PREFIX,
+    MEDIA_TYPE_TO_SEARCH,
+    PODCAST_FEATURES,
+)
+from .helpers import get_config_entries, get_librespot_binary
+from .parsers import parse_album, parse_artist, parse_playlist, parse_podcast, parse_track
+from .podcast_helpers import PodcastManager
+from .streaming import LibrespotStreamer
+
+# Expose functions that Music Assistant expects at module level
+__all__ = ["get_config_entries", "setup"]

```

Comment:
can we keep the "get_config_entries" and "setup" here  in the init file to keep consistent with all providers ?